### PR TITLE
Substituting Quicksort algorithms

### PR DESCRIPTION
**Pull request by TomFischer**
The PR substitutes divers implemented versions of the quicksort algorithm using stl functions.

### CODE REVIEW

**Comment by endJunction** [comment]
Jenkins, test this please.

File: BaseLib/quicksort.h [hunk 1]
```diff
@@ -15,243 +12,97 @@
...
+		perm[i] = data[i].second;
 	}
-};
+}
 
-// specialization for pointer types
-template <typename T1, typename T2>
-class Quicksort <T1*, T2>
+template <typename T1, typename T2 = std::size_t>
+void quicksort(T1* & array, std::size_t beg, std::size_t end, T2* & perm)
```
Comment (norihiro-w) [inline BaseLib/quicksort.h]: is this function still used? if not, why not remove it?

**Comment by norihiro-w** [comment]
:+1: besides a minor comment

File: BaseLib/quicksort.h (hunk 1 above)
Comment (TomFischer) [inline BaseLib/quicksort.h]: Yes, it is used within the NestedDissection algorithms.

Comment (norihiro-w) [inline BaseLib/quicksort.h]: i cannot find it. as far as I checked, previously this was used  in AdjMat.cpp but you have changed it to use std::sort().

Comment (TomFischer) [inline BaseLib/quicksort.h]: See MathLib/LinAlg/Sparse/NestedDissectionPermutation/CRSMatrixReordered.cpp.

Comment (norihiro-w) [inline BaseLib/quicksort.h]: i see. thanks

**Comment by bilke** [comment]
:+1:

**Pull request merged**